source/mir/math/stat.d

```diff
@@ -424,6 +426,241 @@ unittest
     assert(float.max.repeat(3).hmean!(double, "fast").approxEqual(float.max));
 }
 
+/++
+Computes the median of `slice`.
+
+Can also pass a a flag `allowModify` that allows the input slice to be modified.
+By default, a copy is made. 
+
+Returns:
+    the median of the slice
+
+See_also: $(SUBREF mean)
++/
+template median(F, Flag!"allowModify" allowModify = No.allowModify)
+{
+    F median(Iterator, size_t N, SliceKind kind)(Slice!(Iterator, N, kind) slice)
+    {
+        assert(slice.elementCount > 0, "median: slice must have length greater than zero");
+
+        import mir.ndslice.topology: flattened;
+
+        static if (allowModify == No.allowModify) {
+            import mir.ndslice.allocation: rcslice;
+            
+            if (slice.elementCount > 2) {
+                auto val = slice.rcslice.flattened;
+                auto temp = val.lightScope;
+                return medianImpl!F(temp);
+            } else {
+                return medianImpl!F(slice.flattened);
```

Comment:
It doesn't look we need special `medianImpl`, instead, we can call the function itself.

Reply:
The problem is if I keep `allowModify` as a template parameter, as in the simplistic version below:

```
import std.stdio: writeln;

template median(bool allowModify = false) {
    T median(T)(T x) {
        static if (!(allowModify))
            return median!(true)(x);
        else
            return 5;
    }
}

void main() {
    writeln(median(3));
    writeln(median!true(3));
}
```

Reply:
`.` before the function name solves the problem. `return .median!(true)(x);`